src/controllers/dynakube/version/image_hash.go

```diff
@@ -2,80 +2,68 @@ package version
 
 import (
 	"context"
+	"crypto/x509"
 	"fmt"
+	"net/http"
+	"net/url"
 
+	"github.com/Dynatrace/dynatrace-operator/src/api/v1beta1/dynakube"
 	"github.com/Dynatrace/dynatrace-operator/src/dockerconfig"
-	"github.com/containers/image/v5/image"
-	"github.com/containers/image/v5/manifest"
-	"github.com/containers/image/v5/transports/alltransports"
-	"github.com/containers/image/v5/types"
-	"github.com/opencontainers/go-digest"
-	"github.com/pkg/errors"
+	"github.com/Dynatrace/dynatrace-operator/src/dockerkeychain"
+	"github.com/Dynatrace/dynatrace-operator/src/registry"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	// VersionLabel is the name of the label used on ActiveGate-provided images.
-	VersionLabel = "com.dynatrace.build-version"
-)
-
-type ImageVersion struct {
-	Version string
-	Digest  digest.Digest
-}
-
 // ImageVersionFunc can fetch image information from img
-type ImageVersionFunc func(ctx context.Context, imageName string, dockerConfig *dockerconfig.DockerConfig) (ImageVersion, error)
+type ImageVersionFunc func(ctx context.Context, registryClient registry.ImageGetter, imageName string, dockerConfig *dockerconfig.DockerConfig) (registry.ImageVersion, error)
 
 var _ ImageVersionFunc = GetImageVersion
 
 // GetImageVersion fetches image information for imageName
-func GetImageVersion(ctx context.Context, imageName string, dockerConfig *dockerconfig.DockerConfig) (ImageVersion, error) {
-	transportImageName := fmt.Sprintf("docker://%s", imageName)
+func GetImageVersion(ctx context.Context, registryClient registry.ImageGetter, imageName string, dockerConfig *dockerconfig.DockerConfig) (registry.ImageVersion, error) {
+	keychain := dockerkeychain.NewDockerKeychain(dockerConfig.RegistryAuthPath, afero.NewOsFs())
+	transport := http.DefaultTransport.(*http.Transport).Clone()
 
-	imageReference, err := alltransports.ParseImageName(transportImageName)
+	ref, err := name.ParseReference(imageName)
 	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+		return registry.ImageVersion{}, fmt.Errorf("parsing reference %q: %w", imageName, err)
 	}
 
-	systemContext := dockerconfig.MakeSystemContext(imageReference.DockerReference(), dockerConfig)
+	if dockerConfig.Dynakube.HasProxy() {
+		proxyUrl, err := url.Parse(dockerConfig.Dynakube.Spec.Proxy.Value)
```

Comment:
[Proxy.ValueFrom](https://github.com/Dynatrace/dynatrace-operator/blob/main/src/api/v1beta1/dynakube/dynakube_types.go#L43) may also be used to define proxy

Reply:
You are totally right!!! Thank you, I missed it

Reply:
Fixed